packages/indexer-cli/src/commands/indexer/cost/get.ts

```diff
@@ -51,25 +51,24 @@ module.exports = {
     const rawDeployment = ['model', 'variables'].includes(first) ? second : first
 
     try {
-      validateDeploymentID(rawDeployment)
-    } catch (error) {
-      print.error(error.toString())
-      process.exitCode = 1
-      return
-    }
-
-    const config = loadValidatedConfig()
-    const deployment = parseDeploymentID(rawDeployment)
-
-    // Create indexer API client
-    const client = await createIndexerManagementClient({ url: config.api })
-    try {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const [identifier, identifierType] = await processIdentifier(rawDeployment, { all: true, global: false })
+      if (identifierType === SubgraphIdentifierType.SUBGRAPH || (identifierType === SubgraphIdentifierType.GROUP && identifier !== 'all')){
+        print.error('Cost models currently must use deployment identifiers. Subgraph identifier for cost models must be of type deployment ID.')
+        process.exitCode = 1
+        return
+      }
+      const config = loadValidatedConfig()
+      const deployment = parseDeploymentID(identifier)
```

Comment:
maybe I can move process identifier and identifier type checks into this function?

Reply:
That sounds good.